app/models/miq_provision_workflow.rb

```diff
@@ -4,13 +4,15 @@ def self.base_model
   end
 
   def self.class_for_platform(platform)
-    "MiqProvision#{platform.classify}Workflow".constantize
+    "MiqProvision#{platform.classify}Workflow".safe_constantize ||
+      "ManageIQ::Providers::#{platform.classify}::CloudManager::ProvisionWorkflow".safe_constantize ||
+      "ManageIQ::Providers::#{platform.classify}::InfraManager::ProvisionWorkflow".constantize
   end
 
   def self.class_for_source(source_or_id)
-    source = source_or_id.kind_of?(ActiveRecord) ? source_or_id : VmOrTemplate.find_by_id(source_or_id)
+    source = VmOrTemplate.find_by_id(source_or_id)
     return nil if source.nil?
-    class_for_platform(source.class.model_suffix)
+    source.ext_management_system.class.provision_workflow_class
```

Comment:
There is a use case with service_templates that will fail here because the new reliance on `ext_management_system`.  A service_template can have a relationship to a provisioning template that identifies the source object to provision.  If the source becomes archived or orphaned (loses it's relationship to the EMS) we still want to be able to load the workflow and allow a user to select a new source object. (PR #2947)

Is there a way to determine the platform value from the just the source object?  I would not hold up this PR for that, but it will need to be addressed at some point soon.